File: l3/z2/Szesciokat.cpp
#include "Szesciokat.hpp"
#include <cmath>

namespace Zadanie {
    Szesciokat::Szesciokat(float a, std::string figureName)
            : mA(a), Figura(std::move(figureName)) {
    }

    float Szesciokat::area() {
        return static_cast<float>((3 * std::pow(mA, 2) * std::sqrt(3)) / 2);
    }

    float Szesciokat::length() {
        return 6 * mA;
    }

    bool Szesciokat::validate(float a) {
        if (a > 0)
            return true;
        else
            return false;
    }

}

File: l3/z1/Szesciokat.java
public class Szesciokat extends Figura{
    Szesciokat(float a) {
        this.mA = a;
        this.mFigureName = "sześciokąt";
    }

    @Override
    public float area() {
        return (float)((3 * Math.pow(mA, 2) * Math.sqrt(3)) / 2);
    }

    @Override
    public float length() {
        return 6 * mA;
    }

    static boolean validate(float a) {
        return a > 0;
    }

    public String getFigureName() {
        return mFigureName;
    }


    private final float mA;

    private final String mFigureName;
}
